proposals/wg-image-compatibility.md

```diff
@@ -0,0 +1,106 @@
+
+Operating systems can use different kernel versions with different configurations.
+Different operating system distributions may integrate different devices and their corresponding drivers,
+even for the same device different versions of drivers may be used. Therefore, different operating system distributions are not identical.
+Even within the same OS distro, different releases are also not identical.
+
+An image compatibility specification would help container image authors describe compatibility requirements in a standard way.
+The specification will be uploaded with the image to the image registry. This makes hard container compatibility requirements discoverable,
+programmable, and will support different consumers and cover use cases where the application requires a specific compatible environment.
+
+For further reference please follow:
+* [OCI Image Compatibility - the spec reasoning](https://docs.google.com/presentation/d/1F9GnCm2sULuyTJ5BEFZlL8Qjab81DK7g9Oy6qBfe5Qs/edit?usp=sharing)
+* [Revisit container and host OS compatibility](https://docs.google.com/document/d/1lzwh8DGMu5vXXHwJmnewYIMffkcOEvH8owX4UYjRcw0/edit#heading=h.fcxt9vheg92c)
+
+## Scope
+
+* Define image compatibility spec that describes special host OS requirements of containerized application.
+* The spec should describe container requirements for Linux, FreeBSD and Windows.
+* The final list of supported fields (like kernel modules, configuration, out-of-tree drivers etc.)
+  is to be determined by the working group for each supported OS.
+* Make the change as minimally invasive as possible to the image, runtime and distribution specs - that could be achieved over
+  delivering compatibility artifact type.
+* Deprecate os.version and os.features fields from the image spec that were mainly used for Windows.
+
+## Out of Scope
+
+* Change images build or container execution flow.
+* Introduce a new hard requirement to build images or run containers.
+
+## Intended work product
+
+* The result of this WG should be extended or a new OCI specification that allows containers to describe host OS requirements.
+* The specification will provide a much better experience for users to find out about container requirements against host OS, if those exist.
+* The specification will provide programmatic data for various OS operators that could influence configuration of nodes.
+
+## Proposed Owners
+
+The following have agreed to participate in the working group, review progress, report progress back to the OCI community, and present the results to the TOB once the working group has completed its objectives.
+
+* Marcin Franczyk (@mfranczy)
+* Joe Huang (@chaoyihuang)
+* Zvonko Kaiser (@zvonkok)
+
+## Stakeholders
+
+OCI Projects, non-OCI projects, or organizations sponsoring the working group and participating in the implementation and use case validation of the work done by the group.
+
+* Huawei
+* NVIDIA
```

Comment:
What does it mean to get added to this list? Are there CLAs involved or similar?

Reply:
As to my knowledge, it only means that company or organization that you work for agreed for your contribution and benefit out of the project. There are no CLAs involved or similar.

@sudo-bmitch could you confirm if the above is correct? Thank you.

Reply:
I've been using this as a list of likely adopters and implementers of the change. E.g. if we were changing registries, I'd look for a representative sample of registry operators. For this, it would be good to see a variety of runtime implementations.

Reply:
The thing we want to achieve here is that we can perform validation without influencing image or runtime spec. Our plan is to decouple from runtime engines and make validation self sufficient, which will provide better decisions making in terms of containers scheduling, policies management, nodes reconfiguration etc.

To make the above realistic, we are going to provide SDK for different set of software like k8s controllers, policy framework (e.g. Themis for Nephio), standalone cli tools or operators that reconfigure nodes.

It is important that validation happens before runtime. That's the reason I think it should not be a hard requirement to have variety of runtime representatives, but of course they are welcome to join!